src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,110 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    public static final int DEFAULT_CAPACITY = 10;
+    private T[] data;
+    private int size;
+
+    public ArrayList() {
+        data = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == data.length) {
+            growTo(getNewCapacity(size));
+        }
+        data[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Incorrect index: " + index);
+        }
+        if (size == data.length) {
+            growTo(getNewCapacity(size));
+        }
+        T[] temporaryData = (T[]) new Object[data.length];
+        System.arraycopy(data, 0, temporaryData, 0, index);
+        temporaryData[index] = value;
+        System.arraycopy(data, index, temporaryData, index + 1, size - index);
```

Comment:
you dont need additional array to move elements

Reply:
Thanks for the tip